src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from 'react';
+
+import * as RadixCheckbox from '@radix-ui/react-checkbox';
+import { IconCheck } from '../Icons';
+
+import classNames from 'classnames';
+import styles from './Checkbox.module.scss';
+const cx = classNames.bind(styles);
+
+export interface LinkProps {
+  text: string;
+  onClick?: () => void;
+}
+export interface CheckboxProps {
+  className?: string;
+  name: string;
+  text?: string;
+  isDisabled?: boolean;
+  checked?: boolean;
+  link?: LinkProps;
+  variant?: 'Standard' | 'Advanced';
+  onChange?: (value: boolean) => void;
+}
+
+export const Checkbox = ({
+  className,
+  name,
+  text,
+  link,
+  isDisabled,
+  checked,
+  variant = 'Standard',
+  onChange
+}: CheckboxProps) => {
+  const [isChecked, setIsChecked] = useState(checked);
+
+  const handleChange = (checkState: boolean) => {
+    if (!isDisabled) {
+      setIsChecked(checkState);
+      onChange(checkState);
+    }
+  };
+
+  return (
+    <div className={cx(className, styles.ZuiCheckbox, styles[`ZuiCheckbox${variant}`])} data-disabled={isDisabled}>
+      <RadixCheckbox.Root
+        className={styles.CheckboxRoot}
+        data-checked={isChecked}
+        checked={isChecked}
+        disabled={isDisabled}
+        onCheckedChange={handleChange}
+        id={name}
+      >
+        <RadixCheckbox.Indicator className={styles.CheckboxIndicator}>
+          <IconCheck size={'0.9rem'} />
+        </RadixCheckbox.Indicator>
+      </RadixCheckbox.Root>
+      {(text || link) && <Label name={name} link={link} text={text} isDisabled={isDisabled} />}
+    </div>
+  );
+};
+
```

Comment:
nit: can we get a comment separating these? [Check here](https://github.com/zer0-os/zApp-NFTs/blob/main/src/features/view-nft/NFTDetailsCard/NFTDetailsCard.tsx#L62-L64) to see what I mean! It helps with readability

Reply:
Added the comment label & link